File: app/src/main/java/sk/styk/martin/apkanalyzer/util/SharedPreferencesHelper.java
package sk.styk.martin.apkanalyzer.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by Martin Styk on 06.11.2017.
 */

public class SharedPreferencesHelper {

    SharedPreferences sharedPreferences;

    public SharedPreferencesHelper(Context context) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public void putString(String key, String value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public String readString(String key) {
        return sharedPreferences.getString(key, null);
    }

    public void putBoolean(String key, Boolean value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public boolean readBoolean(String key) {
        return sharedPreferences.getBoolean(key, false);
    }

    public boolean readBoolean(String key, boolean defValue) {
        return sharedPreferences.getBoolean(key, defValue);
    }

}

File: app/src/main/java/sk/styk/martin/apkanalyzer/util/SharedPreferencesHelper.kt
package sk.styk.martin.apkanalyzer.util

import android.content.Context
import android.preference.PreferenceManager

/**
 * @author Martin Styk
 * @version 06.11.2017.
 */

class SharedPreferencesHelper(context: Context) {

    internal var sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

    fun putString(key: String, value: String) {
        val editor = sharedPreferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun readString(key: String): String? {
        return sharedPreferences.getString(key, null)
    }

    fun putBoolean(key: String, value: Boolean?) {
        val editor = sharedPreferences.edit()
        editor.putBoolean(key, value!!)
        editor.apply()
    }

    fun readBoolean(key: String): Boolean {
        return sharedPreferences.getBoolean(key, false)
    }

    fun readBoolean(key: String, defValue: Boolean): Boolean {
        return sharedPreferences.getBoolean(key, defValue)
    }

    fun putLong(key: String, value: Long) {
        val editor = sharedPreferences.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun readLong(key: String): Long {
        return sharedPreferences.getLong(key, -1)
    }

    fun readLong(key: String, defValue: Long): Long {
        return sharedPreferences.getLong(key, defValue)
    }

}
